components/constellation/constellation.rs

```diff
@@ -799,17 +802,16 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
-    fn add_pending_change(&mut self, change: SessionHistoryChange)
-    {
-        self.handle_load_start_msg(change.new_pipeline_id);
+    fn add_pending_change(&mut self, change: SessionHistoryChange) {
+        self.handle_load_start_msg(change.top_level_browsing_context_id, change.new_pipeline_id);
         self.pending_changes.push(change);
     }
 
     /// Handles loading pages, navigation, and granting access to the compositor
     #[allow(unsafe_code)]
     fn handle_request(&mut self) {
         enum Request {
-            Script(FromScriptMsg),
+            Script((PipelineId, FromScriptMsg)),
```

Comment:
This is not consistent with `NetworkListener`. Also, it's going to make the following select map a bit more complex.